contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorExtractor.java

```diff
@@ -16,122 +16,32 @@
 
 package io.opentelemetry.contrib.trace.propagation;
 
-import static io.opentelemetry.contrib.trace.propagation.B3Propagator.COMBINED_HEADER;
-import static io.opentelemetry.contrib.trace.propagation.B3Propagator.COMBINED_HEADER_DELIMITER;
 import static io.opentelemetry.contrib.trace.propagation.B3Propagator.MAX_SPAN_ID_LENGTH;
 import static io.opentelemetry.contrib.trace.propagation.B3Propagator.MAX_TRACE_ID_LENGTH;
-import static io.opentelemetry.contrib.trace.propagation.B3Propagator.SAMPLED_HEADER;
-import static io.opentelemetry.contrib.trace.propagation.B3Propagator.SPAN_ID_HEADER;
-import static io.opentelemetry.contrib.trace.propagation.B3Propagator.TRACE_ID_HEADER;
 import static io.opentelemetry.contrib.trace.propagation.B3Propagator.TRUE_INT;
 
 import io.grpc.Context;
 import io.opentelemetry.context.propagation.HttpTextFormat;
 import io.opentelemetry.internal.StringUtils;
-import io.opentelemetry.trace.DefaultSpan;
 import io.opentelemetry.trace.SpanContext;
 import io.opentelemetry.trace.SpanId;
 import io.opentelemetry.trace.TraceFlags;
 import io.opentelemetry.trace.TraceId;
 import io.opentelemetry.trace.TraceState;
-import io.opentelemetry.trace.TracingContextUtils;
-import java.util.Objects;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import javax.annotation.concurrent.Immutable;
 
 @Immutable
-final class B3PropagatorExtractor {
+abstract class B3PropagatorExtractor {
```

Comment:
What do think about making this an interface, rather than an abstract class?

Reply:
Okay. If we make them interface then, will we be moving `buildSpanContext`, `isTraceIdValid` and `isSpanIdValid ` to Util class I guess. Not sure if we want to make them `default` method of the interface.

Reply:
You can still have static methods on your interface.

Reply:
I believe it is Java 8 and above right? I don't think we have static and default in Java 7.

Reply:
oh shoot, you're right. Since they are only used in one place, you can have an inner util class in the interface, like we do in the `SpanData` right now, rather than pulling it out of the interface entirely.

Reply:
sounds good